File: problemas/1100/1179.cs
using System;
using System.Collections.Generic;

namespace uri1179
{
    internal static class Program
    {
        private static void Main()
        {
            var pares = new List<int>();
            var impares = new List<int>();

            for (int i = 0; i < 15; i++)
            {
                int n = int.Parse(Console.ReadLine());

                if (impares.Count == 5)
                {
                    ImprimeNumeros(impares, false);
                }
                else if (pares.Count == 5)
                {
                    ImprimeNumeros(pares, true);
                }

                if (n % 2 == 0)
                {
                    pares.Add(n);
                }
                else
                {
                    impares.Add(n);
                }
            }

            if (impares.Count > 0)
            {
                ImprimeNumeros(impares, false);
            }

            if (pares.Count > 0)
            {
                ImprimeNumeros(pares, true);
            }
        }

        private static void ImprimeNumeros(List<int> numeros, bool saoPares)
        {
            for (int i = 0; i < numeros.Count; i++)
            {
                Console.WriteLine($"{(saoPares ? "par" : "impar")}[{i}] = {numeros[i]}");
            }

            numeros.Clear();
        }
    }
}

File: problemas/1100/1179.php
<?php

$pares = [];
$impares = [];

for ($i = 0; $i < 15; $i++) {
    $n = readline();

    if (count($impares) == 5) {
        imprimeNumeros($impares, false);
    } else if (count($pares) == 5) {
        imprimeNumeros($pares, true);
    }

    if ($n % 2 == 0) {
        array_push($pares, $n);
    } else {
        array_push($impares, $n);
    }
}

if (count($impares) > 0) {
    imprimeNumeros($impares, false);
}

if (count($pares) > 0) {
    imprimeNumeros($pares, true);
}

function imprimeNumeros(&$numeros, $saoPares) {
    for ($i = 0; $i < count($numeros); $i++) {
        echo ($saoPares ? 'par' : 'impar') . "[{$i}] = {$numeros[$i]}\n";
    }

    $numeros = array();
}